rust/private/rustc.bzl

```diff
@@ -754,18 +748,24 @@ def _compute_rpaths(toolchain, output_dir, dep_info):
     Returns:
         depset: A set of relative paths from the output directory to each dependency
     """
-    if not dep_info.transitive_dylibs:
+    preferreds = [get_preferred_artifact(lib) for lib in dep_info.transitive_noncrates.to_list()]
+
+    # TODO(augie): I don't understand why this can't just be filtering on
```

Comment:
Captain Obvious speaking, but you could just print(dylibs) and build/run the failing example and compare the content?

Reply:
Yeah, I tried that and it didn't make any sense.

Reply:
Ok, this PR is already touching many sensitive pieces, let's defer fixing this into a followup PR.